packages/cli/init/src/utils/loadOpenApiFromUrl.ts

```diff
@@ -0,0 +1,43 @@
+import axios from "axios";
+import { writeFile } from "fs/promises";
+import { dump } from "js-yaml";
+import { join } from "path";
+import tmp from "tmp-promise";
+import { Logger } from "../../../logger/src/Logger";
+
+export enum LoadOpenApiStatus {
+    Failed,
+    Success
+}
+
+class LoadOpenApiResult {
+    public msg: string;
+    public status: LoadOpenApiStatus;
+    public constructor(message = "", status: LoadOpenApiStatus) {
+        this.msg = message;
+        this.status = status;
+    }
+}
```

Comment:
I would use a discriminated union for this 
```
export interface LoadOpenAPIResult = SuccessOpenAPIResult | FailureOpenAPIResult; 

export interface SuccessOpenAPIResult {
  type: "success", 
  path: ""
}

export interface FailedOpenAPIResult {
  type: "failure", 
  errorMessage: ...
}
```

Reply:
this will force the user to visit success of error before being able to use those proeprties (can read more here about typescript discriminated unions: https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html)

Reply:
Ahh yes, that makes it much cleaner :)